Update OSX install instructions

rgozim opened the pull request:
OSX install instructions for getting started with OMERO without using the OMERO Homebrew formula. 

The guide contains a similar structure to what was present before. 

Here is an outline of following sections that have changed the most:

- **Requirements**: 
    Instructs which Homebrew packages OMERO requires and guides the user into using 
    Virtualenv.
- **Installation**: 
    - Instructs the user on how to download pre-built version of OMERO and configure OS X
    - Instructs the user on how to download and build OMERO from source and configure OS X
- **Configuration**: 
    Instructs how to start a Virtualenv terminal and configure OMERO

rgozim committed:
Updated homebrew instructions

rgozim committed:
Updated install instructions

rgozim committed:
Further update, adding insturctions to setup a bash_profile

rgozim committed:
Further changes

rgozim committed:
Updated osx installation docs

rgozim committed:
Tweaked osx installation guide

rgozim committed:
Some format changes

rgozim committed:
Added missing enviorment variables

jburel commented:
@rgozim not included in the build
One conflict to fix

rgozim committed:
Added missing colon

joshmoore commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-and Python`_ page. Note that the automated script linked above tests
-the OMERO installation using the Homebrew Python.
+3. OMERO depends on ICE 3.6 and unfortuently does not run with 
+   the latest version of ICE at this time (ICE 3.7.3). To obtain 
```
:+1: but do we need a reminder somewhere? Will we grep for `3.6` and `3.7` globally for OMERO 5.5?

mtbc replied:
usually "Ice" (even though I think it's an acronym!)
(also below)

joshmoore commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-    $ brew install omero53 --with-nginx --with-cpp
-    $ export PYTHONPATH=$(brew --prefix omero53)/lib/python
-    $ export ICE_CONFIG=$(brew --prefix omero53)/etc/ice.config
+    $ mkdir -p ~/Omero
```
This may be confusing considering omero itself creates `~/omero`.

joshmoore commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-    $ which omero
-    /usr/local/bin/omero
+    $ curl https://downloads.openmicroscopy.org/omero/5.3.4/artifacts/OMERO.server-5.3.4-ice36-b69.zip > ~/Omero/server.zip
```
I think we have variables for these strings

joshmoore commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-Set up OMERO data directory::
+    $ which omero
+    /Omero/server/bin/omero
```
Missing `~`

joshmoore reviewed (commented):
Few initial comments

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-Requirements
+
+OSX Basics
```
We usually space out "OS X" though admittedly we're not wholly consistent.
(also below)

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-Requirements
+
+OSX Basics
 ------------
```
Underline length should match title length.

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 ------------
 
-.. _`Homebrew wiki`: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Installation.md
+In order to develop on OMERO, we recommend you ensure you have your MAC setup for
```
I don't think we ever all-caps "MAC".

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-.. _`Homebrew wiki`: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Installation.md
+In order to develop on OMERO, we recommend you ensure you have your MAC setup for
+developement. The first step to achieving this is to create a .bash_profile file in the
```
development

mtbc replied:
`:file:` markup at least for first introduction of this

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-    $ ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
-    $ export PATH=/usr/local/bin:$PATH
+To open your .bash_profile in a text editor, such as the built in TextEdit app, use::
```
hyphenate "built-in" as it's an adjectival phrase

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-    $ brew update
-    $ brew doctor
+.. note::
+   If you want to see changes to your .bash_profile take affect without restarting
```
"effect"

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-.. _`Homebrew and Python`: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Homebrew-and-Python.md
+   To ensure PostgreSQL uses UTF-8 encoding, open your bash profile and 
+   add the following enviorment variables::
```
environment

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-more thorough description of the Homebrew solution, see the `Homebrew
-and Python`_ page. Note that the automated script linked above tests
-the OMERO installation using the Homebrew Python.
+3. OMERO depends on ICE 3.6 and unfortuently does not run with 
```
unfortunately

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
+    $ brew install zeroc-ice/tap/ice36
+
+.. Note::
+   If you already have a version of ICE thats not 3.6 installed, you can instruct Homebrew to *unlink* it using 
```
"...that is not..." (try to avoid contractions)

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
     $ brew install python
 
-Check that Python is working and is version 2.7::
+   Homebrew installs Python in the follwing location::
```
following

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
+    export PATH="/usr/local/opt/python/libexec/bin:$PATH"
+
+.. note:: 
+   To keep things a little cleaner, add the following enviorment variable to your .bash_profile::
```
environment

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-https://github.com/Homebrew/homebrew-science for the HDF5 formula and
-https://github.com/ome/homebrew-alt for all the OME-provided formulae
-and older versions of Ice. To add these, run::
+6. For developing with OMERO, or Python in general, we would recommend you use VirtaulEnv.
```
typo in "virtualenv", also don't think I've seen that capitalization before

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-and older versions of Ice. To add these, run::
+6. For developing with OMERO, or Python in general, we would recommend you use VirtaulEnv.
+   VirtualEnv allows us to develop python applications without having to 
+   worry about clashing thirdparty packages for different Python projects.
```
hyphenate "third-party"

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-https://github.com/ome/homebrew-alt for all the OME-provided formulae
-and older versions of Ice. To add these, run::
+6. For developing with OMERO, or Python in general, we would recommend you use VirtaulEnv.
+   VirtualEnv allows us to develop python applications without having to 
```
initial capital for "Python", especially as it has one on the next line

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
+
+   This will create a folder to hold Python libraries in the the directory :file:`~Virtual/omero/lib`
+
+.. note:: You can activate the VirtualEnv enviorment that we created using::
```
environment

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-See the :download:`step01_deps.sh <walkthrough/osx/step01_deps.sh>` script for
-the steps described above.
+      Reload bash_profile in OSX::
```
above we've been including the "." prefix

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
+^^^^^^^^^^^^^^^^
 
-To install and deploy the |release| release of OMERO.server, run::
+1. Using the command-line Terminal, prepare a place for your OMERO server to 
```
Not sure why terminal now has an initial capital? It didn't above. (Gets one again below though.)

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-    $ cd /usr/local
-    $ bash bin/omero_python_deps
+2. Once extracted, open your .bash_profile in a text editor, 
+   such as the built in TextEdit app::
```
built-in

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
+    $ open -a TextEdit.app ~/.bash_profile
 
-    $ pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log -w start
+   Add an enviorment variable OMERO_SERVER to the .bash_profile which points
```
"environment", also `:envvar:` markup

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
 
-    $ pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log -w start
+   Add an enviorment variable OMERO_SERVER to the .bash_profile which points
+   to the location of the OMERO executbale::
```
executable (also below)

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -41,155 +51,196 @@ running::
-    $ export ROOT_PASSWORD=${ROOT_PASSWORD:-omero}
-    $ omero db script --password $ROOT_PASSWORD -f - | psql -h localhost -U db_user omero_database
+   To ensure OMERO is correctly linked into your OSX :envvar:`PATH`, type the following in Terminal and ensure
+   you get a similar output::
```
:+1: for these confirmatory steps

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -202,14 +253,72 @@ code from the project's GitHub account to build locally::
+
+    $ source ~/.bash_profile
+
+2. To use Omero, we need to first setup Postgres. Open a command-line terminal and run the 
```
"set up"

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -268,6 +377,31 @@ See example script for a basic functionality test: :download:`step04_test.sh <wa
+Example .bash_profile
+^^^^^^^^^^^^^^^^^^^^^^
+
+Open your .bash_profile in your favourite text editor, such as the built in TextEdit app::
```
built-in

mtbc commented on omero/sysadmins/unix/server-install-homebrew.txt:
```diff
@@ -268,6 +377,31 @@ See example script for a basic functionality test: :download:`step04_test.sh <wa
+
+    $ open -a TextEdit.app ~/.bash_profile
+
+If you've followed this guide your bash profile should look similar to the follwoing::
```
"If you have followed this guide then your ..."

mtbc replied:
"following"

rgozim committed:
Numerous Fixes
Improved consistency
Fixed typos
Improved formatting of items such as enviornment variables

rgozim committed:
Added alias to example .bash_profile

pwalczysko commented:
@rgozim : Conflict here to be fixed. Will go ahead and test the workflow on snapper.

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -24,6 +23,17 @@ Homebrew requires the latest version of Xcode. Install :program:`Xcode` and
+
+.. _`Homebrew wiki`: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Installation.md
+
+Homebrew will install all packages under :file:`/usr/local`. See also: Installation instructions on the `Homebrew wiki`_.
```
The Homebrew wiki link https://github.com/Homebrew/legacy-homebrew/blob/master/share/doc/homebrew/Installation.md points to an archived page, I think it is a good opportunity now to replace it with https://github.com/Homebrew/brew/blob/master/docs/Installation.md, which is where the archived page suggests to go.

rgozim replied:
Well spotted. I'll update it just now.

rgozim committed:
Updated homebrew installion link

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
 
-Install git if not already present::
+.. note::
+   If you want to see changes to your :file:`.bash_profile` take effect without restarting
```
``...to see changes ... take effect..`` -> maybe ``...want changes to your :file:`.bash_profile` take effect...`` (the original sentence seems unnecessary complicated with the ``see``)

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
 
-    $ brew update
-    $ brew doctor
+    $ open -a TextEdit.app ~/.bash_profile
```
this commands fails on snapper with 
```
open -a TextEdit.app ~/.bash_profile
LSOpenURLsWithRole() failed for the application /Applications/TextEdit.app with error -10810 for the file /opt/hudson/.bash_profile.
snapper:~ hudson$ 
```
maybe expected as I am ssh-ed in ? (cc @sbesson )

pwalczysko replied:
@rgozim : I would leave the bit with TextEdit.app completely out. As we see, in some cases it does not work. In general, one should know how to edit a textfile in their system.

pwalczysko replied:
Actually, backing up on my words. This line works fine on a normal Mac. It is just the ssh-ed in snapper which is throwing the error. Can leave as is.

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
+    export LANG=en_US.UTF-8
+    export LANGUAGE=en_US:en
+
+3. OMERO depends on Ice 3.6 and unfortunately does not run with 
```
``OMERO depends on Ice 3.6 and unfortunately does not run with the latest version of Ice at this time (Ice 3.7.3). To obtain Ice 3.6, we need to add a *tap* to Homebrew::`` 
replace with 
``OMERO depends on Ice but does not run with the latest version of Ice at this time (Ice 3.7.3). To obtain the version of Ice recommended for OMERO (Ice 3.6) we need to add a "tap" to Homebrew::``
 (do not like the "unfortunately", and OMERO does not depend on Ice 3.6 specifically, we support also Ice 3.5)

jburel replied:
```
brew info ice
ice: stable 3.7.0 (bottled)
```

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
+
+    '/usr/local/opt/python/libexec/bin'
+
+   Follow the instructions from the brew Python install and set the Homebrew version of Python 
```
Need to reformulate this. I guess you actually do not want me to follow the instructions and add the line ``export PATH="/usr/local/opt/python/libexec/bin:$PATH"`` to my .bash_profile. Instead, you suggest I do what is written in the "Note" below and do https://github.com/openmicroscopy/ome-documentation/pull/1766/files#diff-38c0f35540194fd0f8261d5f85fae3fcR121 and https://github.com/openmicroscopy/ome-documentation/pull/1766/files#diff-38c0f35540194fd0f8261d5f85fae3fcR125. You should make it clear that this is the mainline of the installation, if this is so, not let me do both :)

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
+    export PATH="/usr/local/opt/python/libexec/bin:$PATH"
+
+.. note:: 
+   To keep things a little cleaner, add the following enviornment variable to your :file:`.bash_profile`::
```
Maybe: ``To keep things a little cleaner, you might want to add the following environment variable... instead of adding the line ``export PATH="/usr/local/opt/python/libexec/bin:$PATH"`` above...`` (see my previous comment, you seem to let me do both - is this intended ?)

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
+    export PATH="$PYTHON_BREW:$PATH"
+
+5. Check that Python is working and is version 2.7.x::
 
```
Add line ``source ~/.bash_profile`` - no harm, and would spare me time wondering why it did not work the first time

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
-https://github.com/Homebrew/homebrew-science for the HDF5 formula and
-https://github.com/ome/homebrew-alt for all the OME-provided formulae
-and older versions of Ice. To add these, run::
+6. For developing with OMERO, or Python in general, we would recommend you use Virtualenv.
```
``...we would recommend you use Virtualenv`` -> ``...we recommend the use of Virtualenv.``

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
-https://github.com/ome/homebrew-alt for all the OME-provided formulae
-and older versions of Ice. To add these, run::
+6. For developing with OMERO, or Python in general, we would recommend you use Virtualenv.
+   Virtualenv allows us to develop Python applications without having to 
```
``...allows us to develop....`` -> remove ``us``

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
+   This will create a folder to hold Python libraries in the the directory :file:`~Virtual/omero/lib`
+
+.. note:: You can activate the Virtualenv enviornment that we created using::
+
```
so this Note is not optional, whereas the Note below is optional. I find this confusing. 
Would actually prefer following (I even think that the other OMERO installation Walkthroughs stick to this): 
Compulsory (non-optional) steps should not be in Notes (also see my comment above).

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
+        $ startVmOmero
+
+.. Note::
+   See the :download:`step01_deps.sh <walkthrough/osx/step01_deps.sh>` script for the steps described above.
```
I do not see the steps 

``brew update`` 
and 
``brew doctor``
which are included in the script in this walkthrough

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
 
-The default version of Ice installed by the OMERO formula is currently
-Ice 3.6.
+Local built server
```
Local -> Locally

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -202,14 +253,72 @@ code from the project's GitHub account to build locally::
+
+    $ export PATH=~/code/projects/openmicroscopy/dist/bin:$PATH
 
+   and follow the steps for setting up the database and OMERO data directory as mentioned in the previous section.
```
What previous section ? there is no such section in this doc, which would deal with OMERO data directory creation and setting up the database.

pwalczysko replied:
In fact, I cannot find any reference in the doc which would lead me now to the steps of creating a new local db and /OMERO directory at all. this is definitely missing here

pwalczysko replied:
As discussed, the section is actually below (not "previous") and is called OMERO configuration.

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -202,14 +253,72 @@ code from the project's GitHub account to build locally::
+
+5. Install Python dependencies using pip::
+
+    $ pip install -r ~Omero/server/share/web/requirements-py27-all.txt
```
This line is not working for me.
```
omero) snapper:openmicroscopy hudson$ pip install -r ~Omero/server/share/web/requirements-py27-all.txt
Could not open requirements file: [Errno 2] No such file or directory: '~Omero/server/share/web/requirements-py27-all.txt
```

pwalczysko replied:
this path to the file needs to be corrected

pwalczysko replied:
In the end, the command
``/Projects/Omero/code/openmicroscopy/dist/share/web/requirements-py27-all.txt``
worked for me, but note that we did some moving of directories. (see also https://github.com/openmicroscopy/ome-documentation/pull/1766/files#r140265914)

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
 
-    $ mkdir -p ~/code/projects
-    $ cd ~/code/projects
+    $ mkdir -p ~/Projects/Omero/code/projects
```
@rgozim you wanted to remove the projects subfolder here

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,155 +51,196 @@ running::
 
 .. note::
     If you have a GitHub account and you plan to develop code for OMERO, you
     should make a fork into your own account and then clone this fork to your
```
As discussed, the command below does not clone, so the description here is confusing.

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -202,14 +253,72 @@ code from the project's GitHub account to build locally::
+
+1. Start the database server::
+
+    $ pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log -w start
```
Does not work for me. From the log:
``could not open directory "pg_tblspc": No such file or directory``

rgozim committed:
Numerous changes to formatting and removed errors

rgozim committed:
Updated error

pwalczysko commented:
Getting the ``pillow`` messages on snapper, [trello card](https://trello.com/c/mB6ipfY9/771-pillow-error-message) created

pwalczysko commented:
@sbesson : We are done with the first iteration of this installation walkthrough on snapper. Now we just need to get into the state where we were before we started this installation. We also probably need a remote connection via something lik ``vnc`` application to see the omero webclient in a browser. Thank you very much.

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,175 +51,280 @@ running::
 
-    $ pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log -w start
+   Add an enviornment variable :envvar:OMERO_SERVER to the :file:`.bash_profile` which points
+   to the location of the OMERO executabale::
```
Typo: ``executabale`` -> ``executable`` (one ``a`` too much)

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,175 +51,280 @@ running::
 
-Create database and user::
+    # OMERO Server distribution directory
+    export OMERO_SERVER=~/Projects/Omero/server
```
This does not work as written, as there is no subdirectory called ``/server``. Maybe write in the above text "Replace the OMERO.server.xxx with the name of your unzipped OMERO.server directory." and exchenge the ``server`` -> ``OMERO.serverxxx`` here

pwalczysko replied:
Which does not work for me anyway. So this needs one more revision.

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,175 +51,280 @@ running::
-If you wish to build OMERO.server from source for development
-purposes, using the git repository, first use Homebrew to install the
-OMERO dependencies::
+    $ pip install -r ~Omero/server/share/web/requirements-py27-all.txt
```
And this did not find the ``requirements-py27-all.txt`` for me either

pwalczysko replied:
there are I think 2 errors in the path here: first, after the tilder, a slash is missing, second, the first subdir in the home directory is Projects, not Omero.

pwalczysko replied:
third, as indicated https://github.com/openmicroscopy/ome-documentation/pull/1766/files#r140455644, after unzipping I do not have any ``server`` directory, but ``OMERO.serverxxx`` directory

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
````diff
@@ -41,175 +51,280 @@ running::
+   **(Optional)** To make life easier, you can add an ```alias``` to your :file:`.bash_profile`::
+    
+    # Start Virtualenv for OMERO
+    alias startVmOmero="source ~/Virtual/Omero/bin/activate"
````
I am being offered to create this alias already for the second time. The first time it was https://github.com/openmicroscopy/ome-documentation/pull/1766/files#diff-38c0f35540194fd0f8261d5f85fae3fcR163

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,175 +51,280 @@ running::
+
+6. Create and run script to initialize the OMERO database::
+
+    $ omero db script --password omero -f - | psql -h localhost -U db_user omero_database
```
@rgozim : You wanted to remove the OMERO.web section below, as it is not the development server default and confuses matters.

pwalczysko replied:
But for that, we need to add to the config above the lines dealing with web server being develop, see https://github.com/openmicroscopy/ome-documentation/pull/1766/files#r140526104

pwalczysko commented on omero/sysadmins/unix/server-install-homebrew.rst:
```diff
@@ -41,175 +51,280 @@ running::
+    $ omero config set omero.data.dir ~/Projects/Omero/data
+    $ omero config set omero.db.name omero_database
+    $ omero config set omero.db.user db_user
+    $ omero config set omero.db.pass db_password
```
Add lines 
```
omero config set omero.web.application_server development
omero config set omero.web.debug True
```
I do not see why we should instruct developers to use non-develpment web server, @sbesson @will-moore

will-moore replied:
I think this is probably OK. This install page is not explicitly for developers (although it will be mostly used by devs) and there is the link to how to set up development server.

joshmoore replied:
> I do not see why we should instruct developers to use non-develpment web server, 

My 2cents (though not of high priority for this PR) is that we should be encouraging as _little_ use of Django's single-threaded development server as possible.

will-moore replied:
Hmmm - I still use the Django dev server for all my work. Works for me (mostly)

joshmoore replied:
For one-off's, I certainly do start up the dev server now and then, but for having a feel for the reactivity of the application, it would be worth using nginx in front of OMERO.web frequently. If you see any particular barriers to being able to do so, it would be good to hear what they are.

will-moore replied:
So, I just went to try running nginx locally, starting with looking at the docs, but I notice that we don't have them?! sysadmins/unix/server-install-homebrew.html page just says 
```
Basic setup for OMERO using NGINX:

$ export HTTPPORT=${HTTPPORT:-8080}
$ omero web config nginx-development --http $HTTPPORT > $(brew --prefix omero53)/etc/nginx.conf
See installation script step03_nginx.sh

For detailed instructions on how to deploy OMERO.web in a production environment such as NGINX please see OMERO.web administration.
```
But that links to sysadmins/unix/install-web.html which doesn't have ANY instructions on how to deploy web.
So that's my first barrier!

mtbc replied:
Maybe leave it in then for this PR but plan to guide more toward the nginx way after 5.4.0?

joshmoore replied:
Considering all the work that took part on the OSX install/docs including _dropping_ `--with-nginx` that seems potentially negligent.

joshmoore replied:
For reference, here are the steps that I can use locally:

```
mv /usr/local/etc/nginx/nginx.conf /usr/local/etc/nginx/nginx.conf.orig
bin/omero web config nginx-development > /usr/local/etc/nginx/nginx.conf
nginx -t
nginx
bin/omero web start
```

jburel replied:
minimally those steps should be included
@will-moore will you be okay to open a new PR?

will-moore replied:
To continue the "barriers to using nginx" conversation (maybe should be continued elsewhere), I find when developing OMERO.figure, I need each page refresh to load fresh ```figure.js``` from ```/Users/wmoore/Desktop/FIGURE/figure/omero_figure/static/figure/figure.js```, not from ```/Users/wmoore/Desktop/OMERO/openmicroscopy/dist/lib/python/omeroweb/static/figure/figure.js``` without having to ```$ bin/omero web restart``` each time.

Strangely I currently have
```
$ omero config get omero.web.application_server
wsgi-tcp
```
and run ``` $ omero web start``` in the background.
I can then test web at ```http://localhost:4080/``` **without** nginx running. I guess this must be using the Django development server?

Also, I like being able to add ```print``` statements in Django code and see them shown in the terminal when I run ```$ omeroweb/manage.py runserver```.

joshmoore replied:
>  I guess this must be using the Django development server?

Not exactly. Django has started multiple processes (as opposed to the development servers single) and you are accessing it directly. I've not tried this, but I'd assume it's minimally better than using the development server.

> Also, I like being able to add print statements in Django code and see them shown in the terminal when I run $ omeroweb/manage.py runserver.

I'd encourage more use of `logging`. Then you'd need to just `tail -f` the log file rather than watch the stdout of `runserver`.

joshmoore replied:
> To continue the "barriers to using nginx" conversation (maybe should be continued elsewhere)

Happy to reply on the appropriate document, card, or issue.

will-moore replied:
continued at https://trello.com/c/VdQ4HAl6/65-using-nginx-in-development ....

pwalczysko commented:
Still few things to do here, mainly sorting out the paths in the downloaded server section https://github.com/openmicroscopy/ome-documentation/pull/1766#discussion_r140455644 and https://github.com/openmicroscopy/ome-documentation/pull/1766#discussion_r140457548
and few typos and reformulations as well as pointing rather to development web server https://github.com/openmicroscopy/ome-documentation/pull/1766#discussion_r140526104
and a few typos and reformulations to fix as well.

hflynn commented:
Assuming this can be closed as all the commits are included in #1780
